plugins/slack_app/app/models/slack_message.rb

```diff
@@ -0,0 +1,116 @@
+class SlackMessage
+  def initialize(deploy)
+    @deploy = deploy
+  end
+
+  def deliver
+    url = DeployResponseUrl.find_by(deploy_id: @deploy.id).try(:response_url)
+    Faraday.new(url).post do |req|
+      req.headers['Content-Type'] = 'application/json'
+      req.body = ActiveSupport::JSON.encode message_body
+    end if url.present?
+  end
+
+  def message_body
+    ret = if @deploy.waiting_for_buddy?
+      waiting_for_buddy_body
+    elsif @deploy.failed? || @deploy.errored?
+      failed_body
+    elsif @deploy.succeeded?
+      succeeded_body
+    else
+      running_body
+    end
+    ret['response_type'] = 'in_channel'
+    ret
+  end
+
+  private
+
+  def waiting_for_buddy_body
+    {
+      text: title_str,
+      attachments: [
+        {
+          text: pr_str,
+          callback_id: @deploy.id,
+          attachment_type: 'default',
+          actions: [
+            button(':+1: Approve', 'yes')
+          ]
+        }
+      ]
+    }
+  end
+
+  def running_body
+    {
+      text: title_str,
+      attachments: [
+        {
+          text: pr_str + "\nDeploying…",
```

Comment:
When approving a deploy, we usually look at the change set and risks. It will be good to include the risks in the summary directly on Slack. If displaying the full risk section is too verbose, we could at least show how many risk items are included in this deploy with a link to point to the section.